### PR TITLE
Add Chat feature, Completes FR4.2 and half of FR4.1(Chat only)

### LINKED ISSUE #54: Implement Chat Functionality in Communication Service
FR4.1 The system should allow users to communicate via text and voice call or video call to train their communication skills.
Top priority in week 8

### PR DESCRIPTION
**Pull request by ngzhenteng**
Adds Chat functionality between matched users
- only users belonging to the same "room_id" can speak to each other.
- users will get access to list of pre-defined questions according to their interviewer/interviewee status
- Other services(Like collab) can fire the 'join room' event(refer to ChatWindow.js screenshot below for required params) then listen on the socket 'interviewer switch event' event. The 2nd screenshot below shows the params attached with the fired event 

image
image

Added Chisern's local dev instructions to README.md for easy reference

Tested on local dev with docker compose
Yet to see if it works on deployment :P

Potential problems:
1. Similar to User service, if replica>1, socket may not work correctly

Refinements:
1. UI Has to be properly placed alongside code editor(Sorry i have tried but its taking too much time, i thought it'll be better to get the help of the experts(you guys haha, ive asked amanda for help on the layout of the chat box)
2.  Blocking user from sending further messages when other user has left the room

Fixes #54

### CODE REVIEW

File: comm-service/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+app.options('*', cors())
+
+app.get('/', (_, res) => res.send('Hello World from communication-service'))
+
+const httpServer = createServer(app);
+const io = new Server(httpServer, {
+  path: "/api/comm/chat"
+});
+
+console.log(process.env.JWT_ACCESS_SECRET)
```
Comment (ngchisern) [inline comm-service/index.js:23]: Did you forget to remove this? haha

File: comm-service/controller/comm-controller.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const interviewerSwitchRequestEvent = 'request interviewer switch';
+
+export function registerChatHandlers(io, clientSocket) {
+    clientSocket.on('join room', eventData => {
+        if (!eventData.room_id) {
+            console.log("Join room event emitted without room_id");
+            sendJoinRoomFail(clientSocket, { message: "missing room_id" });
+        }
+        const room_id = eventData.room_id;
+        const username = eventData.username;
```
Comment (ngchisern) [inline comm-service/controller/comm-controller.js:15]: In case you didn't know, we are able to get the username directly from the socket.

**Review by ngchisern — APPROVED** [review]
LGTM

File: comm-service/index.js (hunk 1 above)
Comment (ngzhenteng) [reply comm-service/index.js:23]: ahaah yes i did

File: comm-service/controller/comm-controller.js (hunk 2 above)
Comment (ngzhenteng) [reply comm-service/controller/comm-controller.js:15]: Yep thank you! For now i shall not change the implementation, but can do so in the future

**Pull request merged**